crates/reward/src/lib.rs

```diff
@@ -223,31 +224,30 @@ impl<T: Config> Pallet<T> {
         Ok(reward)
     }
 
-    pub fn withdraw_stake(
-        currency_id: T::CurrencyId,
-        account_id: &DefaultVaultId<T>,
-        amount: SignedFixedPoint<T>,
-    ) -> Result<(), DispatchError> {
-        if amount > Self::stake(currency_id, account_id) {
-            return Err(Error::<T>::InsufficientFunds.into());
-        }
+    pub fn withdraw_stake(vault_id: &DefaultVaultId<T>, amount: SignedFixedPoint<T>) -> Result<(), DispatchError> {
+        for currency_id in [vault_id.wrapped_currency(), T::GetGovernanceTokenCurrency::get()] {
+            if amount > Self::stake(currency_id, vault_id) {
+                return Err(Error::<T>::InsufficientFunds.into());
+            }
 
-        checked_sub_mut!(Stake<T>, currency_id, &account_id, &amount);
-        checked_sub_mut!(TotalStake<T>, currency_id, &amount);
+            checked_sub_mut!(Stake<T>, currency_id, &vault_id, &amount);
+            checked_sub_mut!(TotalStake<T>, currency_id, &amount);
```

Comment:
Same here, please make the appropriate changes so we don't duplicate stake storage per `currency_id`.